src/Data/LLVM/BitCode/IR/Metadata.hs

```diff
@@ -672,29 +739,29 @@ parseMetadataEntry vt mt pm (fromEntry -> Just r) =
-        <*>                                parseField r 9 numeric         -- dispScopeLine
-        <*> (mdForwardRefOrNull ctx mt <$> parseField r 10 numeric)       -- dispContainingType
-        <*>                                parseField r 11 numeric        -- dispVirtuality
-        <*>                                parseField r 12 numeric        -- dispVirtualIndex
+        <$> (mdForwardRefOrNull ctx mt <$> parseField r 1 numeric)             -- dispScope
+        <*> (mdStringOrNull     ctx pm <$> parseField r 2 numeric)             -- dispName
+        <*> (mdStringOrNull     ctx pm <$> parseField r 3 numeric)             -- dispLinkageName
+        <*> (mdForwardRefOrNull ctx mt <$> parseField r 4 numeric)             -- dispFile
+        <*>                                parseField r 5 numeric              -- dispLine
+        <*> (mdForwardRefOrNull ctx mt <$> parseField r 6 numeric)             -- dispType
+        <*>                                pure isLocal                        -- dispIsLocal
+        <*>                                pure isDefinition                   -- dispIsDefinition
+        <*>                                parseField r (7 + offsetA) numeric  -- dispScopeLine
+        <*> (mdForwardRefOrNull ctx mt <$> parseField r (8 + offsetA) numeric) -- dispContainingType
+        <*>                                pure virtuality                     -- dispVirtuality
+        <*>                                parseField r (10 + offsetA) numeric -- dispVirtualIndex
         <*> (if hasThisAdjustment
-            then parseField r 19 numeric
-            else return 0)                                                -- dispThisAdjustment
-        <*> parseField r 13 numeric                                       -- dispFlags
-        <*> parseField r 14 nonzero                                       -- dispIsOptimized
-        <*> (optFwdRef hasUnit       15)                                  -- dispUnit
-        <*> (optFwdRef (not hasUnit) (adj 15))                            -- dispTemplateParams
-        <*> (mdForwardRefOrNull ctx mt <$> parseField r (adj 16) numeric) -- dispDeclaration
-        <*> (mdForwardRefOrNull ctx mt <$> parseField r (adj 17) numeric) -- dispVariables
-        -- Indices 18-19 seem unused.
-        <*> (optFwdRef hasThrownTypes 20)                                 -- dispThrownTypes
+            then parseField r (16 + offsetB) numeric
+            else return 0)                                                      -- dispThisAdjustment
+        <*> pure diFlags                                                        -- dispFlags
+        <*> pure isOptimized                                                    -- dispIsOptimized
+        <*> (optFwdRef hasUnit (12 + offsetB))                                  -- dispUnit
+        <*> (mdForwardRefOrNull ctx mt <$> parseField r (13 + offsetB) numeric) -- dispTemplateParams
```

Comment:
In lines 755-759, the delta with `offsetB` against the old value is 3, but for lines 760-762, the delta is 2, then back to 3 again in line 763.  Just want to confirm that this is the intended set of offset base values.

Reply:
Yeah, this algorithm is nuts.  I double checked against the LLVM parser code, as far as I can tell, I've copied the logic accurately.